buildbot/buildbot/worker.py

```diff
@@ -226,6 +230,7 @@ def stage_make(db, config):
             prefix + [
                 'make',
                 'ESTIMATE={}'.format(task['estimate']),
+                'DIRECTIVES={}'.format(task['directives']),
```

Comment:
I assume you probably want to not pass anything at all to the Makefile when `task['directives']` is empty? Otherwise, this will overwrite a `DIRECTIVES` that's defined in the Makefile already, and you probably want that to be filled with the "correct" directives name by default.

Reply:
That's a great point. I didn't think of that. 
So, I'm going to change it to the following,
One can submit a directive file to buildbot, buildbot will run the make command with `DIRECTIVES` set to the file name. Otherwise, it'll run it as before.
I might still be missing something from our chat on Friday.
Is there a case to say `DIRECTIVES=0` or `1`?

Reply:
Hmm... that sounds a little more complicated than I was imagining. Here’s what I was suggesting:

- The Makefile has `DIRECTIVES := foo` or whatever in it already.
- The way you use this is that just typing `make` without any arguments at all will use directives from foo. If you want to tell the thing not to use directives, do `DIRECTIVES=` on the command line.
- Buildbot passes nothing by default. Only when `DIRECTIVES` is set to something does it pass something to `make` on the command line.

Easy! No need for automatic file detection.

Reply:
Ah yes, that sounds great! Thanks!